### ISSUE TITLE
Delayed Start

### ISSUE THREAD

**Issue by quietMichael** [arkmanager/ark-server-tools#1001]
Hi is it possible to start the Arkserver staggered at start @ all or updates? The server z.b. to be started in 5 minutes Server 1 Starts Server 2 Starts 5min Later?

Greetings Michael

**Comment by BoiseComputer** [comment]
I will be looking to implement that in the next few days. I was just discussing this with someone, so perfect timing. What I will most likely do is have it wait until it sees the main instance as available and then start the other instances. Same thing with updates. We will need to stop all, update the main, and then do a start @all.

**Comment by quietMichael** [comment]
That would be perfect since my root server does not start shank it all at once is a delayed start after each other much better do i get a message when the time comes?

greetings

**Comment by BoiseComputer** [comment]
What I am coding in right now, is to start the defaultinstance first. Wait for it to become available and then start the rest immediately. It gives feedback on the screen of what step it is at.

**Comment by quietMichael** [comment]
What exactly do you mean by the main instance? I've got Pro Root 2 server running, that would mean in my case yes one will start then the other will be delayed properly?

**Comment by BoiseComputer** [comment]
In the atlasmanager.cfg there is a line that looks like this.
defaultinstance="main"
You would set the value that is currently main, to whatever the instance name is for your primary node. Then it will always start that first and wait for it to load when using the start @all command.

**Comment by quietMichael** [comment]
I use the ARK tool for ARK server there is this entry is not it still compatible?

**Comment by BoiseComputer** [comment]
This will be fully compatible between both systems. Just manage ARK  with the command arkmanager and ATLAS with the atlasmanager command. I am currently running 5 ark instances and 4 atlas instances side by side.

**Comment by quietMichael** [comment]
How do I have to insert this? Is there any date to replace or is there an update tool?

**Comment by BoiseComputer** [comment]
Once I finish implementation, you can run "atlasmanager  upgrade-tools" and it will automatically update the script.

**Comment by BoiseComputer** [comment]
Ah. Didn't realize I was in the ARK Server Tools thread. I made this code for the ATLAS Server Tools. I will suggest it and if they would like to use it they can. I have it ready, but can't guarantee they will accept it. Sorry for the misunderstanding.

**Comment by quietMichael** [comment]
This is a pity: / maybe one of the creators of these codes can embed it in the ARK tools?

**Comment by BoiseComputer** [comment]
Yes. I submitted the code to them a little bit ago. If they accept it, it will be in the ARK tools code.

**Comment by quietMichael** [comment]
Ok then I hope you will install this :) Thank you very much

**Comment by BoiseComputer** [comment]
They accepted the feature, so you should be able to update and use it now.

**Comment by klightspeed** [comment]
It looks like we need to fix up the ordering, as bash associative arrays are unordered, meaning that the neither keep the original order nor sort by key.

**Comment by quietMichael** [comment]
I do not understand that, is it possible to update the ARKmanager again? And is there a guide on how to enter what and where to start it?

**Comment by klightspeed** [comment]
v1.6.44 includes the fixed instance ordering.

**Comment by quietMichael** [comment]
Now the following error occurs when I try @all to start / usr / local / bin / arkmanager: Line 220: [: Missing `] 'besides it repeats the delay several times.

**Comment by quietMichael** [comment]
He repeats with me several times the delay Starts the first server then he starts the delay after expiration he restarts the delay after the 3rd delay he starts the server first

**Comment by quietMichael** [comment]
quiet@of-stein01:~$ arkmanager start @all
Running command 'start' for instance 'scorched'
Checking for updates before starting
Checking for update; PID: 17104
Your server is already up to date! The most recent version is 3475476.
The server is starting...
The server is now running, and should be up within 10 minutes
Waiting up to 300 seconds before starting next instance .................................................................................................................................................................................................................................................................................................
Running command 'start' for instance 'The_Island'
Checking for updates before starting
Checking for update; PID: 25097
Your server is already up to date! The most recent version is 3475476.
The server is starting...
The server is now running, and should be up within 10 minutes
Waiting up to 300 seconds before starting next instance .....................................................................................................................................................................................................................................................................................
Running command 'start' for instance 'Scorched_Earth_P'
Checking for updates before starting
Checking for update; PID: 1485
Your server is already up to date! The most recent version is 3475476.
The server is starting...
The server is now running, and should be up within 10 minutes
Waiting up to 300 seconds before starting next instance ...........
Running command 'start' for instance 'theisland'
Checking for updates before starting
Checking for update; PID: 2236
Your server is already up to date! The most recent version is 3475476.
The server is starting...
The server is now running, and should be up within 10 minutes
Waiting up to 300 seconds before starting next instance .................

**Comment by klightspeed** [comment]
What is the output of `arkmanager list-instances`?
It sounds like you have `configfile_*` pointing to instance files with different names.

**Comment by quietMichael** [comment]
configfile_theisland="/***/arkmanager/instances/theisland.cfg"
configfile_scorched="/***/arkmanager/instances/scorched.cfg"


defaultinstance="scorched"

**Comment by klightspeed** [comment]
Did you only just add those `configfile_*` entries to your config?

If so, naming them with the same base name as your instance files will fix this.

**Comment by quietMichael** [comment]
So it should look like this?

configfile_theisland="/***/arkmanager/instances/theisland.cfg"
configfile_scorched="/***/arkmanager/instances/scorched.cfg"

defaultinstance="scorched"

**Comment by klightspeed** [comment]
Yes

**Comment by klightspeed** [comment]
I'll fix it up so that it won't delay if it's the last instance.

**Comment by quietMichael** [comment]
Ok that would be great then I am waiting for the update but he will be the first then start a delay and then the last 2 in my case?

**Comment by klightspeed** [comment]
There will be a delay after each instance started if 2 or more instances are being started.

**Comment by quietMichael** [comment]
Great :) Then I'm waiting for the update

**Comment by quietMichael** [comment]
Another question if now z.b. a mod or ARK update comes the script ends automatically with stop @ all and starts again with start @ all?

**Comment by quietMichael** [comment]
Hi, have another problem discovered 1.er Updatet the server is not independent. And if you z.b. arkmanager restart @all makes, then he makes the delay not synonymous if you stop the server with arkmanager stop @ all and again with arkmanager start @ all he starts no delay.

**Comment by klightspeed** [comment]
At the moment updates will restart the server instance if it needs to be updated.  It will not currently stop all server instances before applying the update because normally each server lives in its own directory.

With v1.6.44b, `arkmanager restart @all` should be equivalent to `arkmanager stop @all start @all` - i.e. it should stop all of the servers before starting them again, with a delay between each server starting.

**Comment by BoiseComputer** [comment]
I would think reasoning behind not automatically restarting all after an update is because you could configure each of your ark instances to use their own install base. Then only the one would require a restart. Perhaps a solution is to look at all instances that have the same path set for arkserverroot=. You could then restart all instances that have the exact same root since they would require an update. Just a thought.

**Comment by quietMichael** [comment]
I noticed the update of the Arkmanager the ARK servers start automatically at the root reboot or start. However, there without a delay .. Can you still install?

greetings

**Comment by klightspeed** [comment]
By default when not doing a user-install, the arkmanager installer will install a service to start it (named arkmanager).  You can disable this if you want to start it another way.

Note that the service will start all of the instances in parallel.
